src/DynamicData/List/Internal/Transformer.cs

```diff
@@ -98,15 +98,31 @@ private void Transform(ChangeAwareList<TransformedItemContainer> transformed, IC
                 case ListChangeReason.Replace:
                     {
                         var change = item.Item;
-                        Optional<TDestination> previous = transformed[change.PreviousIndex].Destination;
-                        if (change.CurrentIndex == change.PreviousIndex)
+
+                        if (change.CurrentIndex == -1 || change.PreviousIndex == -1)
```

Comment:
@RolandPheasant  Would it ever happen that the CurrentIndex is valid but the PreviousIndex is missing?  If so, then I might need to tweak this a bit so that it follows the code path at the bottom that allows the currentIndex to be different than the previous index.

I'm not sure that's a valid concern though.

Reply:
It's possible, but if they are out of sync then probably best to place it safe and assume their both in a bad state.